### PR TITLE
Show full names and check whether review was created by requester

### PR DESCRIPTION
**Pull request by tyll**
This contains some commits from PR #66 since they are needed to get pkgdb-admin running. Not sure how to best create PR that depend on each other.

### CODE REVIEW

**Comment by pypingou** [comment]
So, should we rebase this branch now that #66 was merged?

File: pkgdb2client/utils.py [hunk 1]
```diff
@@ -255,26 +267,37 @@ def check_package_creation(info, bugid, pkgdbclient):
...
         if flag['name'] == 'fedora-review':
             if flag['status'] == '+':
-                flag_setter = flag['setter']
+                flag_setter_email = flag['setter']
+                flag_setter, flag_setter_full = get_fasinfo(flag_setter_email)
                 if is_packager(flag_setter):
                     messages["good"].append(
-                        'Review approved by packager {0}'.format(flag_setter))
+                        'Review approved by packager `{0}`'.format(
+                            flag_setter_full))
```
Comment (ralphbean) [inline pkgdb2client/utils.py]: Should we include their username here also?  i.e.  'Review approved by packager `Ralph Bean (ralph)`'?

File: pkgdb2client/utils.py [hunk 2]
```diff
@@ -255,26 +267,37 @@ def check_package_creation(info, bugid, pkgdbclient):
...
                 else:
                     messages["bad"].append(
-                        'Review approved by non-packager {0}'.format(
-                            flag_setter))
-                if flag_setter == bug.creator:
+                        'Review approved by non-packager `{0}`'.format(
+                            flag_setter_full))
```
Comment (ralphbean) [inline pkgdb2client/utils.py]: Here as well?  I leave it up to you.

**Comment by ralphbean** [comment]
Looks like this has been idle for over a month, and it has merge conflicts now.  :(

@tyll, @pypingou -- what do you want to do?

**Comment by tyll** [comment]
It can rebase it eventually, but looking at it now it seems like the packagedb-cli master branch was rebased... Was this intentional?

**Comment by ralphbean** [comment]
Not from me.

**Comment by pypingou** [comment]
I know I had issues with this repo a little while ago, looks like for projects like this one that I don't touch so often I sometime I sometime forget to push to both repo (here and on fedorahosted) and thus end-up with a broken situation :-/.

File: pkgdb2client/utils.py (hunk 1 above)
Comment (tyll) [inline pkgdb2client/utils.py]: AFAICS, the username is included there

File: pkgdb2client/utils.py (hunk 2 above)
Comment (tyll) [inline pkgdb2client/utils.py]: and here as well - ```flag_setter_full``` contains all informatione.

**Comment by tyll** [comment]
Rebasing it to master was easy, since mostly merge commits vanished. Therfore this should be ready now afaics

File: pkgdb2client/utils.py [hunk 3]
```diff
@@ -187,6 +187,18 @@ def __get_fas_user_by_email(email_address):
...
 
+def get_fasinfo(email):
+    ''' Get fas username and build a name string for the user like:
+        Human Name (fas_username) <email>
+    '''
+
+    fas_user = __get_fas_user_by_email(email)
+    fas_username = fas_user.get("username", "").encode("utf-8")
+    human_name = fas_user.get("human_name", "").encode("utf-8")
+    full_info = "{0} ({1}) <{2}>".format(human_name, fas_username, email)
```
Comment (pypingou) [inline pkgdb2client/utils.py]: There are cases where an email in bugzilla is not linked to a FAS account, these people have a dedicated mapping in python-fedora that's overridden by a hotfix on pkgdb0{1,2} [1]. So I wonder if we shouldn't check what ``fas_user`` is and if it is ``None``, then return a sensible string using the email.

[1] https://infrastructure.fedoraproject.org/infra/ansible/roles/pkgdb2/files/fas2.py

Comment (tyll) [inline pkgdb2client/utils.py]: I hope I addressed it like you meant.

File: pkgdb2client/utils.py [hunk 4]
```diff
@@ -187,6 +187,27 @@ def __get_fas_user_by_email(email_address):
...
 
 
+def get_fasinfo(email):
+    ''' Get fas username and build a name string for the user like:
+        Human Name (fas_username) <email>
+    '''
+
+    fas_user = __get_fas_user_by_email(email)
+
+    fas_username = fas_user["username"]
```
Comment (pypingou) [inline pkgdb2client/utils.py]: This won't work if/when ``fas_user`` is None.

Comment (tyll) [inline pkgdb2client/utils.py]: On Mon, Feb 08, 2016 at 11:38:03PM -0800, Pierre-Yves Chibon wrote:
> > @@ -187,6 +187,27 @@ def __get_fas_user_by_email(email_address):

> > +    fas_user = __get_fas_user_by_email(email)
> > +
> > +    fas_username = fas_user["username"]
> 
> This won't work if/when ``fas_user`` is None.

Thank you, I addressed this now as well.

**Comment by pypingou** [comment]
Ok :+1: for me, I'll merged.

We will just have to make sure that admins know that an email unknown to FAS can be (but is not necessarily) a bad thing or a reason to reject of a request.

**Comment by tyll** [comment]
On Wed, Feb 10, 2016 at 02:44:14AM -0800, Pierre-Yves Chibon wrote:

> We will just have to make sure that admins know that an email unknown to FAS can be (but is not necessarily) a bad thing or a reason to reject of a request.

Can you give an example when it is not a reason to reject a request?
The e-mail addresses from the hotfix seem to work to query FAS. If there
is a bugzilla e-mail address that is not tied to FAS, can you please
tell me which one so I can take a look.

**Comment by pypingou** [comment]
All the emails in the hotfix should correspond to account in bugzilla that have a different email in FAS (I am speaking of the ``self.__bugzilla_email = {`` dictionary in https://infrastructure.fedoraproject.org/infra/ansible/roles/pkgdb2/files/fas2.py just to be sure we speak about the same thing).

**Comment by tyll** [comment]
On Wed, Feb 10, 2016 at 03:16:02PM -0800, Pierre-Yves Chibon wrote:
> All the emails in the hotfix should correspond to account in bugzilla
> that have a different email in FAS (I am speaking of the
> ``self.__bugzilla_email = {`` dictionary in
> https://infrastructure.fedoraproject.org/infra/ansible/roles/pkgdb2/files/fas2.py
> just to be sure we speak about the same thing).

I dug a little deeper. python-fedora also ships with a copy of this list
afaics, therefore as long as python-fedora is update to date
packagedb-cli will handle all these special e-mail addresses correctly.

**Comment by pypingou** [comment]
Except that python-fedora is in fact not up to date :)

**Comment by pypingou** [comment]
Anyway like I said, I'm ok w/ merging this, we should just warn the admins running it that there might be false-positive sometime.

**Pull request merged**